fedimint-server/src/consensus/mod.rs

```diff
@@ -239,18 +253,26 @@ impl FedimintConsensus {
 
         funding_verifier.verify_funding()?;
 
+        self.tx_sender
+            .send(transaction)
+            .await
+            .map_err(|_e| TransactionSubmissionError::TxChannelError)?;
+        Ok(())
+    }
+
+    /// For saving a tx, should be done prior to making a consensus proposal
+    pub async fn save_transaction_to_db(&self, transaction: Transaction) {
+        let mut dbtx = self.db.begin_transaction().await;
+
         let new = dbtx
-            .insert_entry(&ProposedTransactionKey(tx_hash), &transaction)
+            .insert_entry(&ProposedTransactionKey(transaction.tx_hash()), &transaction)
             .await
             .expect("DB error");
         dbtx.commit_tx().await.expect("DB Error");
```

Comment:
Follow up: We should probably make every `commit_tx` an `autocommit` call eventually.

Reply:
Why is that?

Reply:
Always better to report an error and retry then crash, IMO.